### PR TITLE
Refactor Route Drawer

### PR DESCRIPTION
**Pull request by RoiEXLab**
This makes the UI code more independent from the calculation code...

### CODE REVIEW

File: src/main/java/games/strategy/triplea/ui/MapRouteDrawer.java [hunk 1]
```diff
@@ -39,12 +41,13 @@
   public static final double DETAIL_LEVEL = 1.0;
   private static final int arrowLength = 4;
 
-  private final RouteOptimizer routeOptimizer;
+  private final RouteCalculator routeOptimizer;
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ui/MapRouteDrawer.java]: Should this field be renamed to `routeCalculator`?

File: src/main/java/games/strategy/triplea/ui/logic/Line.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  public Point getP2() {
+    return new Point(x2, y2);
+  }
+
+  public Line2D toLine2D() {
+    return new Line2D.Double(x1, y1, x2, y2);
+  }
+
+  @Override
+  public boolean equals(Object o) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ui/logic/Line.java]: If you're going to override `equals()`, you probably want to override `hashCode()`, as well, in order to satisfy the contract of `Object`.

File: src/main/java/games/strategy/triplea/ui/logic/Point.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  public double distance(Point point) {
+    return Math.sqrt(Math.pow(this.getX() - point.getX(), 2) + Math.pow(this.getY() - point.getY(), 2));
+  }
+
+  public java.awt.Point toPoint() {
+    return new java.awt.Point((int) x, (int) y);
+  }
+
+  @Override
+  public boolean equals(Object o) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ui/logic/Point.java]: Ditto comment in `Line#equals()`.

File: src/main/java/games/strategy/triplea/ui/MapRouteDrawer.java (hunk 1 above)
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/ui/MapRouteDrawer.java]: Yes, I forgot that ^^
Noticed this as well

**Comment by RoiEXLab** [comment]
Updated...

**Pull request merged**